text/0000-fabric-gateway.md

```diff
@@ -0,0 +1,213 @@
+
+The Gateway also runs as a gRPC server and exposes the following services to client applications (SDKs):
+
+```
+service Gateway {
+    rpc Endorse(ProposedTransaction) returns (PreparedTransaction) {}
+    rpc Submit(PreparedTransaction) returns (stream Event) {}
+    rpc Evaluate(ProposedTransaction) returns (Result) {}
+}
+```
+
+The client credentials (private key) are never passed to the Gateway.  Client applications are responsible for managing their user keys (optionally using SDK Wallets) and signing the protobuf payloads.
+
+Submitting a transaction is a two step process (performed by the client SDK):
+- __Endorse__
+  - The client creates a ProposedTransaction message, which contains a SignedProposal message as defined in `fabric-protos/peer/proposal.proto` and signed with the user's identity.  The ProposedTransaction message also contains other optional properties, such as the endorsing orgs if the client wants to specify that.
+  - The `Endorse` service is invoked on the Gateway, passing the ProposedTransaction message
+    - The Gateway will determine the endorsement plan for the requested chaincode and forward to the appropriate peers for endorsement. It will return to the client a `PreparedTransaction` message which contains a `Envelope` message as defined in `fabric-protos/common/common.proto`.  It will also contain other information, such as the return value of the chaincode function and the transaction ID so that the client doesn't necessarily need to unpack the `Envelope` payload.
+- __Submit__
+  - The client signs the hash of the Envelope payload using the user's private key and sets the signature field in the `Envelope` in the `PreparedTransaction`.
+  - The `Submit` service is invoked on the Gateway, passing the signed `PreparedTransaction` message.  A stream is opened to return multiple return values.
+    - The Gateway will register transaction event listeners for the given channel/txId.
+    - It will then broadcast the `Envelope` to the ordering service.
+    - The success/error response is passed back to the client in the stream
+    - The Gateway awaits suffient tx commit events before returning and closing the stream, indicating to the client that transaction has been committed.
+
+Evaluating a transaction is a simple process of invoking the `Evaluate` service passing a SignedProposal message.  The Gateway passes the request to a peer of it's choosing according to a defined policy (probably same org, highest block count) and returns the chaincode function return value to the client.
+
+## Launching the Gateway
+
+#### Standalone process
+When running standalone, the Gateway is effectively a 'client' application to a set of peers.  It will be configured to connect to a peer in the organization in order to invoke discovery.  To do this it will need a signing identity in the channel writers policy.
```

Comment:
> in the organization 

I assume you're trying to say that the gateway is associated with an organization and it is configured to connect to a peer in the same organization?